### PR TITLE
Reset individual preview state even when not previewing, fixes lp1910814

### PR DESCRIPTION
**Pull request by daschuer**
https://bugs.launchpad.net/mixxx/+bug/1910814

### CODE REVIEW

**Comment by Be-ing** [comment]
We should really have tests for this behavior. We've had several major bugs in the past few weeks.

File: src/engine/controls/cuecontrol.cpp [hunk 1]
```diff
@@ -878,16 +878,14 @@ void CueControl::hotcueActivatePreview(HotcueControl* pControl, double value) {
...
-            pControl->setPreviewing(false);
-            pControl->setPreviewingPosition(Cue::kNoPosition);
-
-            // If this is the last hotcue to leave preview.
+    } else if (pControl->isPreviewing()) {
+        // Mark this hotcue as not previewing.
+        double position = pControl->getPreviewingPosition();
+        pControl->setPreviewing(false);
+        pControl->setPreviewingPosition(Cue::kNoPosition);
+        if (m_iCurrentlyPreviewingHotcues) {
```
Comment (uklotzde) [inline src/engine/controls/cuecontrol.cpp:886]: Please use `> 0` to indicate that this is not a boolean but a counter. The name is misleading and the `i` prefix is unsuitable to reveal the actual purpose.

**Comment by uklotzde** [comment]
@Be I agree. Both the hot cue and sync code are unmaintainable for anyone who was not involved. If you touch it, it breaks.

Tests that depend on timing are not helpful. Instead the underlying state machines have to be extracted and tested in isolation. Unfortunately, I don't see a chance that this will ever be possible without a clean rewrite. Too many dependencies.

**Comment by uklotzde** [comment]
If I trigger the hot cue again while paused the track keeps playing when releasing the hot cue!?

**Comment by uklotzde** [comment]
Tried to abuse it, works as expected so far. LGTM

**Comment by uklotzde** [comment]
@daschuer Would you please merge this to main? There are non-trivial merge conflicts that need to be resolved.

**Comment by Abdull** [comment]
I have used this PR now for two hours with the latest 2.3 branch and I haven't experienced lp1910814 anymore. Thanks!

**Comment by Be-ing** [comment]
Ping @daschuer can you please merge this to main?

**Pull request merged**